vortex-serde/src/layouts/read/layouts/inline_schema.rs

```diff
@@ -0,0 +1,139 @@
+use std::sync::Arc;
+
+use bytes::Bytes;
+use flatbuffers::root;
+use vortex_dtype::DType;
+use vortex_error::{vortex_err, VortexResult};
+use vortex_flatbuffers::{footer, message};
+
+use crate::layouts::read::cache::{LazyDeserializedDType, RelativeLayoutCache};
+use crate::layouts::{
+    LayoutDeserializer, LayoutId, LayoutReader, LayoutSpec, Message, ReadResult, Scan,
+    INLINE_SCHEMA_LAYOUT_ID,
+};
+use crate::stream_writer::ByteRange;
+
+#[derive(Debug)]
+pub struct InlineSchemaLayoutSpec;
+
+impl LayoutSpec for InlineSchemaLayoutSpec {
+    fn id(&self) -> LayoutId {
+        INLINE_SCHEMA_LAYOUT_ID
+    }
+
+    fn layout(
+        &self,
+        fb_bytes: Bytes,
+        fb_loc: usize,
+        scan: Scan,
+        layout_reader: LayoutDeserializer,
+        message_cache: RelativeLayoutCache,
+    ) -> Box<dyn LayoutReader> {
+        Box::new(InlineSchemaLayout::new(
+            fb_bytes,
+            fb_loc,
+            scan,
+            layout_reader,
+            message_cache,
+        ))
+    }
+}
+
+#[derive(Debug)]
+pub struct InlineSchemaLayout {
+    fb_bytes: Bytes,
+    fb_loc: usize,
+    scan: Scan,
+    layout_builder: LayoutDeserializer,
+    message_cache: RelativeLayoutCache,
+    child_layout: Option<Box<dyn LayoutReader>>,
+}
+
+enum DTypeReadResult {
+    ReadMore(Vec<Message>),
+    DType(DType),
+}
+
+impl InlineSchemaLayout {
```

Comment:
is this meant for arrays that aren't top-level Struct? e.g. I just want to write a VarBinArray out to a file

Reply:
Don't you just want to serialize the schema into the parent layout metadata?

There is almost a strict 1:1 between arrays and layouts, things should work mostly the same way.

Reply:
The current use case is stats tables